src/storage/page_buffer.c

```diff
@@ -2389,6 +2368,29 @@ pgbuf_check_page_ahead_of_replication (THREAD_ENTRY * thread_p, PAGE_PTR page)
 }
 #endif
 
+int
+pgbuf_check_for_deallocated_page_or_desyncronization (THREAD_ENTRY * thread_p, PGBUF_LATCH_MODE latch_mode,
+						      const VPID & vpid)
+{
+  int ret = ER_PB_BAD_PAGEID;
+
+  PAGE_PTR fixed_page = pgbuf_fix (thread_p, &vpid, OLD_PAGE_DEALLOCATED, latch_mode, PGBUF_UNCONDITIONAL_LATCH);
+  if (fixed_page == NULL)
+    {
+      return ER_PB_BAD_PAGEID;
```

Comment:
Maybe this time there's another error.
```suggestion
      ASSERT_ERROR_AND_SET (ret);
      return ret;
```